web/src/handler/types/query.rs

```diff
@@ -0,0 +1,48 @@
+use std::{convert::Infallible, future::Future};
+
+use serde::Deserialize;
+
+use crate::{handler::FromRequest, request::WebRequest};
+
+pub struct Query<T>(pub T);
+
+impl<'a, 'r, S, T> FromRequest<'a, WebRequest<'r, S>> for Query<T>
+where
+    S: 'r,
+    T: for<'de> Deserialize<'de>,
+{
+    type Type<'b> = Query<T>;
```

Comment:
```suggestion
impl<'a, 'r, S, T> FromRequest<'a, WebRequest<'r, S>> for Query<T>
where
    S: 'static,
    T: Universe,
    T: Deserialize<'a>,
{
    type Type<'b> = Query<T::Type<'b>>;
```

Where `Universe` is the infamous trait:
```rust
trait Universe {
    type Type<'a>: 'a;
}
```

But this requires every type to be deserialized to impl Universe; an ergonomic hurdle but mybe overcome by a homebrew derive(XitcaDeserialize) that implements if serde::Deserialize and Universe and should also be applicable to Path extractor.